packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -14,21 +21,61 @@ import expo.modules.image.records.CachePolicy
 import expo.modules.image.records.ContentPosition
 import expo.modules.image.records.ImageTransition
 import expo.modules.image.records.SourceMap
+import expo.modules.kotlin.Promise
 import expo.modules.kotlin.functions.Queues
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
 import expo.modules.kotlin.views.ViewDefinitionBuilder
+import java.io.File
 
 class ExpoImageModule : Module() {
   override fun definition() = ModuleDefinition {
     Name("ExpoImage")
 
-    Function("prefetch") { urls: List<String> ->
-      val context = appContext.reactContext ?: return@Function
+    AsyncFunction("prefetch") { urls: List<String>, cachePolicy: CachePolicy, promise: Promise ->
+      val context = appContext.reactContext ?: return@AsyncFunction false
+
+      var imagesLoaded = 0
+      var failed = false
+
       urls.forEach {
         Glide
           .with(context)
-          .download(GlideUrl(it))
+          .load(GlideUrl(it))
```

Comment:
Have to use `.load()` here if we want to get a memory cache.

Reply:
Can you add a comment about that?